packages/@aws-cdk/custom-resource-handlers/lib/custom-resources/aws-custom-resource-handler/index.ts

```diff
@@ -1,17 +1,277 @@
+  try {
+    event.ResourceProperties.Create = decodeCall(event.ResourceProperties.Create);
+    event.ResourceProperties.Update = decodeCall(event.ResourceProperties.Update);
+    event.ResourceProperties.Delete = decodeCall(event.ResourceProperties.Delete);
+    let data: { [key: string]: string } = {};
+
+    // Default physical resource id
+    let physicalResourceId: string;
+    switch (event.RequestType) {
+      case 'Create':
+        physicalResourceId = event.ResourceProperties.Create?.physicalResourceId?.id ??
+                             event.ResourceProperties.Update?.physicalResourceId?.id ??
+                             event.ResourceProperties.Delete?.physicalResourceId?.id ??
+                             event.LogicalResourceId;
+        break;
+      case 'Update':
+      case 'Delete':
+        physicalResourceId = event.ResourceProperties[event.RequestType]?.physicalResourceId?.id ?? event.PhysicalResourceId;
+        break;
+    }
+    const call: AwsSdkCall | undefined = event.ResourceProperties[event.RequestType];
+    // if there is a call there will always be logging configured -- otherwise, in the event of no call, logging
+    // wasn't configured so just default to existing behavior
+    const logApiResponseData = call?.logApiResponseData ?? true;
+    if (call) {
+      const apiCall = new ApiCall(call.service, call.action);
+
+      let awsSdk: AwsSdk | Promise<AwsSdk> = loadAwsSdk(apiCall.v3PackageName, event.ResourceProperties.InstallLatestAwsSdk);
+
+      console.log(JSON.stringify({ ...event, ResponseURL: '...' }));
+
+      let credentials;
+      if (call.assumedRoleArn) {
+        const timestamp = (new Date()).getTime();
+
+        const params = {
+          RoleArn: call.assumedRoleArn,
+          RoleSessionName: `${timestamp}-${physicalResourceId}`.substring(0, 64),
+        };
+
+        const { fromTemporaryCredentials } = await import('@aws-sdk/credential-providers');
+        credentials = fromTemporaryCredentials({
+          params,
+          clientConfig: call.region !== undefined ? { region: call.region } : undefined,
+        });
+      }
+
+      awsSdk = await awsSdk;
+
+      const flatData: { [key: string]: string } = {};
+      try {
+        const response = await apiCall.invoke({
+          sdkPackage: awsSdk,
+          apiVersion: call.apiVersion,
+          credentials: credentials,
+          region: call.region,
+          parameters: decodeSpecialValues(call.parameters, physicalResourceId),
+          flattenResponse: true,
+        });
+
+        if (logApiResponseData) {
+          console.log('API response', response);
+        }
+
+        flatData.apiVersion = apiCall.client.config.apiVersion; // For test purposes: check if apiVersion was correctly passed.
+        flatData.region = await apiCall.client.config.region().catch(() => undefined); // For test purposes: check if region was correctly passed.
```

Comment:
I'm not completely sure since this is existing logic. Let's err on the side of caution for now -- we know this works so let's leave it as is for now and we can revisit in the future.